cmd/open.go

```diff
@@ -15,17 +25,155 @@ var openCmd = &cobra.Command{
 
 Pass the path to the directory that contains the solution you want to see on the website.
 	`,
-	Args: cobra.ExactArgs(1),
 	RunE: func(cmd *cobra.Command, args []string) error {
+		cfg := config.NewConfig()
+
+		v := viper.New()
+		v.AddConfigPath(cfg.Dir)
+		v.SetConfigName("user")
+		v.SetConfigType("json")
+		// Ignore error. If the file doesn't exist, that is fine.
+		_ = v.ReadInConfig()
+		cfg.UserViperConfig = v
+
+		return runOpen(cfg, cmd.Flags(), args)
+	},
+}
+
+func runOpen(cfg config.Config, flags *pflag.FlagSet, args []string) error {
+	var url string
+	usrCfg := cfg.UserViperConfig
+	trackID, _ := flags.GetString("track")
+	exerciseID, _ := flags.GetString("exercise")
+	teamID, _ := flags.GetString("team")
+
+	if exerciseID == "" {
+		// if no --exercise is given, use original functionality
 		metadata, err := workspace.NewExerciseMetadata(args[0])
```

Comment:
How about extracting `args[0]` into a variable `path` up at the start where we're also extracting from flags?

```go
var path string
if args.length > 0 {
  path = args[0]
}
```

That way down here we can say `if path != "" {`

If we want to be really explicit, we could also have a validation on `path != "" and exerciseSlug != ""` and explain that you need one or the other.